### DISCUSSION TITLE
Attributes on default metrics in asgi OpentelemetryMiddleware

### DISCUSSION THREAD

**Discussion by carson-chapman-sumup** [open-telemetry/opentelemetry-python-contrib#3159] [Q&A] [answered]
I'm using the FastAPIInstrumentor to add metrics to my application. FastAPIInstrumentor's asgi OpentelemetryMiddleware emits quite a few metrics by default: http_server_active_requests, http_server_duration_milliseconds, http_server_request_size_bytes, http_server_response_size_bytes.

My questions are:
1. Is it possible to add custom attributes to these default metrics? They are emitted in [this code](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/8db1479e0d2e8c7d1b122859e891fbaf1757d449/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py#L766) and seem to be using hardcoded attributes.
2. If I can't add attributes to those default metrics, is there an easy way to prevent these histogram metrics from being recorded altogether while still being able to use Opentelemetry metrics for my own custom metrics?

**Comment by emdneto** [comment, marked as answer]
1. The attributes are defined based on the Semantic Conventions and indeed, you can't pass labels to duration histograms defined in the instrumentation package. 
2. You can define views, like [this](https://github.com/open-telemetry/opentelemetry-python/blob/main/docs/examples/metrics/views/drop_metrics_from_instrument.py)

See https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1801
